cmd/azure-config-credentials-injector/main.go

```diff
@@ -57,17 +63,26 @@ func mergeCloudConfig(_ *cobra.Command, args []string) error {
 		return fmt.Errorf("%s env variable should be set up", clientIDEnvKey)
 	}
 
-	azureClientSecret, found := os.LookupEnv(clientSecretEnvKey)
-	if !found {
-		return fmt.Errorf("%s env variable should be set up", clientSecretEnvKey)
+	// Handle error cases for authentication methods. We should configure exactly one method.
+	tenantId, federatedTokenFile, err := lookupWorkloadIdentityEnv(tenantIDEnvKey, federatedTokenEnvKey)
+	azureClientSecret, secretFound := os.LookupEnv(clientSecretEnvKey)
+	if err == nil && secretFound {
+		// Workload identity is available, but client secret is provided too.
+		// If we don't fail here we would use client secret unnecessarily, also we can assume that user intended to use workload identity because all required values are provided.
+		return fmt.Errorf("%s env variable is set along with env variables %s and %s which indicate workload identity auth should be used, this should never happen\n"+
+			"Please consider reporting a bug to CCO team: https://github.com/openshift/cloud-credential-operator/issues", clientSecretEnvKey, tenantIDEnvKey, federatedTokenEnvKey)
```

Comment:
You're right, all bug reports should probably land in `issues.redhat.com`, thank you. 

> i think this is mostly making sense to me, although i do not know the azure credentials system at a deep level. i do have a question about one of the log messages.

Here we're mostly just passing through the values we get from CCO, most of the interesting stuff is done [there](https://github.com/openshift/cloud-credential-operator/pull/523) so that's why I wanted @abutcher to review this PR as well.